spring-cloud-kubernetes-client-discovery/src/main/java/org/springframework/cloud/kubernetes/client/discovery/ConditionalOnSelectiveNamespacesDisabled.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright 2013-2023 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.kubernetes.client.discovery;
+
+import java.util.Optional;
+
+import org.springframework.context.annotation.Condition;
+import org.springframework.context.annotation.ConditionContext;
+import org.springframework.core.type.AnnotatedTypeMetadata;
+
+/**
+ * Conditional that checks if our discovery is _not_ based on selective namespaces, i.e.:
+ * 'spring.cloud.kubernetes.discovery.namespaces' is not set.
+ *
+ * @author wind57
+ */
+public final class ConditionalOnSelectiveNamespacesDisabled implements Condition {
```

Comment:
1. I want to achieve a rather simple thing in this PR: add support for "selective namespaces" (we already have it in fabric8). So give users the possibility to set more than one namespace that they want to apply discovery for, without the need to specify "all-namespaces". Again, such support exists in various other places, like fabric8-discovery, catalog implementation, etc. 

Unfortunately, the code is heavily involved in the k8s-native implementation, because of the `Listers` that this one uses, so coming up with an idea how to implement this without breaking (or re-writing everything) was not easy. I hope my comments will make sense below. 

My idea is around these two conditionals. I want to separate the logic in two pieces:

- when `spring.cloud.kubernetes.discovery.namespaces` is _not_ set. When this happens, one auto-configuration must be active: `KubernetesClientInformerAutoConfiguration`. This is an existing class and the only change it requires is the addition of `@Conditional(ConditionalOnSelectiveNamespacesDisabled.class)`.

- when `spring.cloud.kubernetes.discovery.namespaces` is set, meaning selective namespaces is active. When this happens, `KubernetesClientInformerSelectiveNamespacesAutoConfiguration` auto-configuration is active and this is a new one. It is annotated with `@Conditional(ConditionalOnSelectiveNamespacesEnabled.class)`.

Depending on these two conditionals, only one auto-configuration will be active, either:
`KubernetesClientInformerAutoConfiguration` or `KubernetesClientInformerSelectiveNamespacesAutoConfiguration`.